beacon-chain/forkchoice/types/types.go

```diff
@@ -11,3 +12,10 @@ type ProposerBoostRootArgs struct {
 	CurrentSlot     types.Slot
 	SecondsIntoSlot uint64
 }
+
+// BlockAndCheckpoint to call the InsertOptimisticChain function
+type BlockAndCheckpoint struct {
```

Comment:
Sorry, being picky here : )

```suggestion
type BlockAndCheckpoints struct {
```

Reply:
hahaha I'll commit this, but I was certain you were going to say this. The reason I chose one is that I think we can get rid of the finalized checkpoint soon... but will leave that for a future PR.